x-pack/plugins/security_solution/public/detections/components/endpoint_responder/use_responder_action_data.ts

```diff
@@ -6,53 +6,97 @@
  */
 import type { ReactNode } from 'react';
 import { useCallback, useMemo } from 'react';
-import { useGetEndpointDetails, useWithShowEndpointResponder } from '../../../management/hooks';
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import type { SentinelOneAgentInfo } from '../../../../common/types';
+import type { ResponseActionAgentType } from '../../../../common/endpoint/service/response_actions/constants';
+import {
+  useGetEndpointDetails,
+  useWithShowEndpointResponder,
+  useWithShowResponder,
+} from '../../../management/hooks';
 import { HostStatus } from '../../../../common/endpoint/types';
 import {
-  NOT_FROM_ENDPOINT_HOST_TOOLTIP,
-  LOADING_ENDPOINT_DATA_TOOLTIP,
   HOST_ENDPOINT_UNENROLLED_TOOLTIP,
+  LOADING_ENDPOINT_DATA_TOOLTIP,
   METADATA_API_ERROR_TOOLTIP,
+  NOT_FROM_ENDPOINT_HOST_TOOLTIP,
 } from './translations';
+import { getFieldValue } from '../host_isolation/helpers';
 
 export interface ResponderContextMenuItemProps {
   endpointId: string;
   onClick?: () => void;
+  thirdPartyAgentInfo?: {
+    agentType: ResponseActionAgentType;
+    eventData: TimelineEventsDetailsItem[] | null;
+  };
 }
 
+const getSentinelOneAgentInfo = (
```

Comment:
This is using alert data to construct sentinel agent info that is needed for responder header component. The `last_checkin ` is not the same as we have for endpoint.